From: Khalid Abu Shawarib <khalid.shawarib@gmail.com>
Date: Fri, 30 Jan 2026 01:39:13 +0300
Subject: build: Fix no introspection build

It is giving an error that a file with no name is not found. Could be
a change in newer meson versions. Regardless, use the more obvious way
to initialize empty variables.

Origin: https://gitlab.gnome.org/GNOME/gnome-desktop/-/commit/8f816dc
---
 libgnome-desktop/gnome-bg/meson.build | 5 +++--
 libgnome-desktop/meson.build          | 4 ++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/libgnome-desktop/gnome-bg/meson.build b/libgnome-desktop/gnome-bg/meson.build
index 99c137d..7a8f591 100644
--- a/libgnome-desktop/gnome-bg/meson.build
+++ b/libgnome-desktop/gnome-bg/meson.build
@@ -40,6 +40,8 @@ libgnome_bg = library('gnome-bg-4',
   ],
 )
 
+libgnome_bg_gir = []
+
 if get_option('introspection')
   libgnome_bg_gir = gnome.generate_gir(libgnome_bg,
     sources: [libgnome_bg_headers, libgnome_bg_sources],
@@ -52,9 +54,8 @@ if get_option('introspection')
     symbol_prefix: 'gnome',
     install: true,
   )
-else
-  libgnome_bg_gir = ''
 endif
+
 libgnome_bg_dep = declare_dependency(
   sources: [
     libgnome_bg_gir,
diff --git a/libgnome-desktop/meson.build b/libgnome-desktop/meson.build
index 8518a1c..f4a5d5a 100644
--- a/libgnome-desktop/meson.build
+++ b/libgnome-desktop/meson.build
@@ -129,6 +129,8 @@ if get_option('build_gtk4')
     ],
   )
 
+  libgnome_desktop_base_gir = []
+
   if get_option('introspection')
     libgnome_desktop_base_gir = gnome.generate_gir(libgnome_desktop_base,
       sources: [libgnome_desktop_base_headers, libgnome_desktop_base_sources],
@@ -141,8 +143,6 @@ if get_option('build_gtk4')
       symbol_prefix: 'gnome',
       install: true,
     )
-  else
-    libgnome_desktop_base_gir = ''
   endif
 
   pkg.generate(
